docs(contributing): add issue refinement sub-section (#13430)
Esri/calcite-components: CONTRIBUTING.md

### Issues that cannot be worked on

Certain labels indicate that an issue is not ready for development:

- `1 - in design`: Issues that are going through design consultation. Once designers complete the effort, the `2 - ready for dev` label will be added to the issue, which means a developer can pick up the issue.
- `spike`: Issues that need to research a question or resolve a complex task with uncertain outcomes. Once the spike has been performed a `spike complete` label is added to the issue, which means a developer can pick up the issue.
- `need more info`: Issues that are missing information and/or a clear, actionable description. This can mean we are waiting on a user to provide additional context, we can't reproduce the issue, or further discussion is needed in order to determine a solution.

> - `resolution: refine`: Issues that need refinement due to a lack of actionable information or a change in scope/effort since the last update. See the [Issue Refinement](#issue-refinement) section for more details.

Reviewer: Awesome addition! 💪🏻 

nitpick suggestion to use "refer" as opposed to "see" to account for more audiences:

```suggestion
- `resolution: refine`: Issues that need refinement due to a lack of actionable information or a change in scope/effort since the last update. Refer to the [issue refinement section](#issue-refinement) for more details.
```

Author: Great catch 🎣 Thanks!

> - `resolution: refine`: Issues that need refinement due to a lack of actionable information or a change in scope/effort since the last update. Refer to the [issue refinement section](#issue-refinement) for more details.

Reviewer 2: The descriptions for `need more info` and `resolution: refine` read as very similar. If they have different purposes I think we should clarify the distinction a bit more.

This is likely out of scope for this particular PR, but I wanted to also follow up on the possibility of changing `resolution: refine` to `needs refinement`. Is it possible we only need one label here? 

cc @jcfranco @geospatialem

Author: For more context, here are the lists and counts of Issues for both labels:
- [`need more info`](https://github.com/Esri/calcite-design-system/issues?q=sort%3Aupdated-desc%20is%3Aissue%20is%3Aopen%20label%3A%22need%20more%20info%22): 2 open, 57 closed
- [`resolution: refine`](https://github.com/Esri/calcite-design-system/issues?q=sort%3Aupdated-desc%20is%3Aissue%20is%3Aopen%20label%3A%22resolution%3A%20refine%22): 22 open, 9 closed

Reviewer: Will defer on direction of the labels to @jcfranco. Could be swayed either direction - could see them co-exist together as they are today, or merged into `need more info` as long as we can filter between the issues as-needed.

Reviewer 3: Great discussion. I’m also open to merging these or keeping them separate if it helps simplify. Here’s how I’ve typically seen them used:

**`need more info`**

* used when clarification is needed from the submitter  
* mostly aimed at external contributors  
* typically applied after the issue is in a milestone  
* automatically closed after two weeks with no follow up once the label is applied  

**`resolution: refine`**†

* used when requirements or acceptance criteria need more clarity before moving forward  
* applied early for placeholder issues, or after it’s been added to a milestone  
* mostly aimed at the Calcite team for planning  
* no automatic closing  

† **+1** on renaming if we keep it – it was ~copypasta'd~ inspired by a Stencil repo label

Reviewer 2: Based on your description of the two labels, I think we should add some of that context to the descriptions and rename the `resolution: refine` label.

Author: Thank you all for the discussion on this! 🤜 🤛 

I think the usages are clear enough to keep both, especially with the auto-close workflow. Also agree with renaming `resolution: refine` to `needs refinement`.

WDY'allT about the following label descriptions? Would also update them in GitHub's UI.

```suggestion
- `need more info`: Issues that require more information and/or a clear, actionable description from the submitter. If there is no follow up after two weeks since the label was added, the issue will be automatically closed.
- `needs refinement`: Issues that we support, but need refinement of the scope or acceptance criteria before moving forward, sometimes due a change in scope/effort since the last update. Refer to the [issue refinement section](#issue-refinement) for more details.
```

Reviewer 4: I think the use cases make more sense now. 

This may be out of scope, but without the context of the docs though, I think it would be really easy to choose the wrong tag since the two tags sound synonymous. Would it make sense to change the names of both tags? Maybe `needs more info: submitter` and `needs more info: internal`?

Reviewer: > Thank you all for the discussion on this! 🤜 🤛
> 
> I think the usages are clear enough to keep both, especially with the auto-close workflow. Also agree with renaming resolution: refine to needs refinement.
> 
> WDY'allT about the following label descriptions? Would also update them in GitHub's UI.

@isaacbraun Looking great! A few small nitpick considerations for the `needs refinement` context, WDYT?:

> - `needs refinement`: Issues that ~we~ are **supported**, but need **scope** refinement ~of the scope~ or **updated** acceptance criteria before moving forward, ~sometimes due~ **as a result of** a change in scope ~/~ **or** effort ~since the last update~. Refer to the [issue refinement section](#issue-refinement) for more details.

Reviewer 5: 👆+1

Author: Glad to hear we're on the right path 🎉

@DintaMel I hear your concern. To me the currently proposed names actually communicate the need better, since it is not always going to be internal/external but rather a difference in what needs to be updated/clarified. There might be better names options, but I think the current ones are solid. Would appreciate others' thoughts.

Also, the labels do display the descriptions in the label picker, which should help when unfamiliar users are deciding which makes sense.

@geospatialem That looks great, thanks! Updated suggestion below.

```suggestion
- `need more info`: Issues that require more information and/or a clear, actionable description from the submitter. If there is no follow up after two weeks since the label was added, the issue will be automatically closed.
- `needs refinement`: Issues that are supported, but need scope refinement or updated acceptance criteria before moving forward, as a result of a change in scope or effort. Refer to the [issue refinement section](#issue-refinement) for more details.
```

Reviewer 2: I think this is looking good. I happy to move forward with this and revisit further name changes in the future if needed.

Author: Sounds good! Thank you all for the great discussion! 🤜 🤛

I've updated the descriptions and will move forward with merging. After merge I will update the label name and descriptions in GitHub, and we can discuss further name changes as needed 👍

- `need more info`: Issues that are missing information and/or a clear, actionable description. This can mean we are waiting on a user to provide additional context, we can't reproduce the issue, or further discussion is needed in order to determine a solution.
- `blocked`: Issues that cannot be worked on until a different issue is resolved. The blocking issue may be from an external library (Lit, Storybook, Jest, etc.) or a Calcite Components issue. In the body or comments of a blocked issue, include a link to the blocking issue. To track when an issue is unblocked, add a comment in the blocking issue's body referencing the blocked issue(s). Use the following format for the comment: "Blocked issues: #0000, #0000". List multiple blocked issues by separating them with commas. The format of the issues listed can be issue number only (e.g., #0000), or the full issue URL (e.g., github.com/Esri/calcite-design-system/issues/xxxxx).

> 

Reviewer: Should we add the issue refinement label to this section above since we wouldn't necessarily want work to take place on issues where refinement is needed? And if so, we could link to the new section below.

Author: Ah, yes that's a great idea! 👍

Author: Added a new item here and adjusted the below sections wording accordingly. WDYT?

### Issue Refinement

> During an issue's lifecycle, additional information or scope refinement may be necessary before continuing work. Perhaps the issue does not have enough information to act upon, or the scope/effort has changed since the most recent update. We utilize the `resolution: refine` label to communicate these states. Follow the steps below to ensure the issue is properly refined and updates are communicated appropriately.

Reviewer: WDYT about a bit stronger language to this section? Also if the above comment is integrated into the section above, perhaps this section could be reduced a bit?

```suggestion
When the `resolution: refine` label is added to an issue, additional information or scope refinement is needed. The issue may not have enough information to act upon, or the scope/effort has changed since the most recent update. Follow the steps below to ensure the issue is properly refined and updates are communicated.
```

Reviewer 2: This edit reads more clearly to me. Can we add a comma after "When the `resolution: refine` label is added to an issue"?

Author: With the addition of `resolution: refine` to the "Issues that cannot be worked on", I think this is great. Thanks!

Reviewer: Updated the suggestion to include the comma 🤜🏻🤛🏻

> These steps can be performed by whoever has the most relevant context or relationships with the involved teams: Product Engineers (PEs), Developers, or Designers.

Reviewer: Can this sentence be removed and we dive into the list below instead?

Author: Most likely. I included it thinking that it would prevent any confusion for devs/designers about asking a PE to do this work or doing it themselves, but the way I ended up wording the steps it probably isn't necessary anymore. 👍

> 1. Apply the `resolution: refine` label to the issue.

Reviewer: ```suggestion
1. Where refinement is needed, apply the `resolution: refine` label to the issue.
```

> 2. Add a comment explaining **why** the label was added (e.g., missing requirements/information, unclear scope, etc.), and tag:

Reviewer: nitpick:

```suggestion
2. Add a comment explaining **why** the label was added (e.g., missing requirements/information, unclear scope, etc.), and where appropriate mention via an `@` tag the following:
```

Author: Changed this wording slightly, WDYT?
